src/btree/bt_npos.c

```diff
@@ -0,0 +1,362 @@
+        idx = WT_CLAMP(idx, 0, entries - 1);
+        npos_local -= idx;
+        descent = pindex->index[idx];
+
+        if (read_cache) {
+            /*
+             * In case of eviction, we never want to load pages from disk. Also, page_swap with
+             * WT_READ_CACHE will fail anyway and we'll lose our pointer, so avoid making a call
+             * that will fail.
+             */
+            switch (WT_REF_GET_STATE(descent)) {
+            case WT_REF_DISK:
+            case WT_REF_LOCKED:
+            case WT_REF_DELETED:
+                /* Can't go down from here but it's ok to return this page. */
+                goto done;
+            default: /* WT_REF_MEM, WT_REF_SPLIT */
+                goto descend;
+            }
+            /* Unreachable but it's ok to be here. */
+        } else {
+            /* Not eviction */
+            switch (WT_REF_GET_STATE(descent)) {
+            case WT_REF_LOCKED:
+                if (!LF_ISSET(WT_READ_NO_WAIT)) {
+                    WT_RET(__wt_page_release(session, current, flags));
+                    __wt_sleep(0, 10);
+                    goto restart;
+                }
+                /* Fall through */
+            case WT_REF_DELETED:
+                /*
+                 * Can't go down from here. Return this page and __find_closest_leaf will finish the
+                 * job.
+                 */
+                goto done;
+            default: /* WT_REF_DISK, WT_REF_MEM, WT_REF_SPLIT */
+                goto descend;
+            }
+            /* Unreachable but it's ok to be here. */
+        }
+        /* Unreachable but it's ok to be here. */
+
+descend:
+        /*
+         * Swap the current page for the child page. If the page splits while we're retrieving it,
+         * restart the search at the root.
+         *
+         * On other error, simply return, the swap call ensures we're holding nothing on failure.
+         */
+        if ((ret = __wt_page_swap(session, current, descent, flags)) == 0) {
+            current = descent;
+            continue;
+        }
+        if (read_cache && (ret == WT_NOTFOUND || ret == WT_RESTART))
+            goto done;
+        if (ret == WT_RESTART) {
+            WT_RET(__wt_page_release(session, current, flags));
+            goto restart;
+        }
+        return (ret);
+    }
+done:
+    /*
+     * Because eviction considers internal pages in post-order, returning the root page will
+     * indicate the end of walk. Also, eviction will never evict the root. Also, returning a NULL is
+     * not an error for eviction but a signal to start over. So handle this case individually.
+     */
+    if (read_cache && __wt_ref_is_root(current)) {
+        WT_RET(__wt_page_release(session, current, flags));
+        current = NULL;
+    }
+    *refp = current;
+    return (0);
+}
+
+/*
+ * __wt_page_from_npos --
+ *     Find a page given its normalized position.
+ */
+int
+__wt_page_from_npos(
```

Comment:
Technically this is only called within the btree/ module and at this time should be __wti, no? I can see it possibly expanding it's scope in the future but should we not aim to limit the scope of functions until they need to be wider? That seems in line with what the modularity effort aims to do. Additionally we could then remove this function from s_funcs.list.